tests/testthat/test-length_test_linter.R

```diff
@@ -0,0 +1,32 @@
+test_that("skips allowed usages", {
+  linter <- length_test_linter()
+
+  expect_lint("length(x) > 0", NULL, linter)
+  expect_lint("length(DF[key == val, cols])", NULL, linter)
+})
+
+test_that("blocks simple disallowed usages", {
+  linter <- length_test_linter()
+  lint_msg_stub <- rex::rex("Checking the length of a logical vector is likely a mistake. Did you mean ")
+
+  expect_lint("length(x == 0)", rex::rex(lint_msg_stub, "`length(x) == 0`?"), linter)
+  expect_lint("length(x == y)", rex::rex(lint_msg_stub, "`length(x) == y`?"), linter)
+  expect_lint("length(x + y == 2)", rex::rex(lint_msg_stub, "`length(x+y) == 2`?"), linter)
```

Comment:
What's up with the spaces around the infix operator disappearing in the lint message?

Reply:
that's xml_text() combining across nodes. it's not ideal but more trouble than it's worth to fix imo.

open to other ways of suggesting an alternative that avoids this issue but I was stuck.

Reply:
How about fetching the code from line data?

Reply:
Not a bad idea, but what about

```r
length(
  foo(
    x, y, z
  ) ==
  bar(
    a, b, c
  )
)
````

I think the current code is pretty simple & handles this case. I would revisit only upon user request as I think it's not very common anyway.

Looking at the linked fixes I filed using this PR, one would have used bad spacing in the lint message:

https://github.com/r-devel/r-svn/pull/144/files#diff-470e0ffa86df95f53eafdae7529ba5fdf7f2f029afef998c998784fbd894db87R507

Reply:
See also https://github.com/r-lib/xmlparsedata/issues/34. I'm not sure a general solution is currently possible.